src/target/puya.c

```diff
@@ -99,6 +100,22 @@
 /* DBG */
 #define PUYA_DBG_BASE   0x40015800U
 #define PUYA_DBG_IDCODE (PUYA_DBG_BASE + 0x00U)
+/*
+ * The format and values of the IDCODE register are undocumented but the vendor SDK splits IDCODE into 11:0 DEV_ID and
+ * 31:16 REV_ID.
+ */
+#define PUYA_DBG_IDCODE_DEV_ID_SHIFT 0U
+#define PUYA_DBG_IDCODE_DEV_ID_MASK  0xfffU
+#define PUYA_DBG_IDCODE_REV_ID_SHIFT 16U
+#define PUYA_DBG_IDCODE_REV_ID_MASK  0xffffU
+
+/* On PY32F002AW15U an IDCODE value of 0x60001000 was observed */
+#define PUYA_DEV_ID_PY32F002A 0x000
+/*
+ * On PY32F002BF15P an IDCODE value of 0x20220064 was observed. Internet search shows the same value is used on
+ * PY32F002BW15.
+ */
+#define PUYA_DEV_ID_PY32F002B 0x064
```

Comment:
Please suffix these with `U` so they are defined unsigned.

Reply:
Fixed locally; will push together with the other changes once they are done.